lib/asset/assetmetadata.js

```diff
@@ -29,13 +30,13 @@ class AssetMetadata {
      */
     constructor(filename, contentType, contentLength) {
         if (filename && typeof filename !== "string") {
-            throw Error(`Invalid filename: ${filename}`);
+            throw new IllegalArgumentError("Invalid filename", filename);
         }
         if (contentType && typeof contentType !== "string") {
-            throw Error(`Invalid contentType: ${contentType}`);
+            throw new IllegalArgumentError("Invalid contentType", contentType);
```

Comment:
For this one it would be good to elaborate the error text - because from the text you can't know if the issue is the content type value or that it's not a string.